src/o365/base/DateAndPeriodBasedReport.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  CommandOption, CommandValidate
+} from '../../Command';
+import * as path from 'path';
+import * as fs from 'fs';
+import PeriodBasedReport, { UsagePeriodOptions } from './PeriodBasedReport';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends UsagePeriodOptions {
```

Comment:
Following this definition, options for DateAndPeriodBasedReport assume that period is always specified while in reality, it's optional and the user can provide either date or period. To keep the types in sync with the intent, it would be better to keep these Options independent of the UsagePeriodOptions and define period as optional.

Reply:
Hi @waldekmastykarz,
I did some changes and tried to separate the interfaces. I think I overcomplicated the solution. Let me know what you think. If you feel that some minor changes improve the situation, I do not mind if you are changing it directly.

Thank you for your feedback,
Patrick